### PR TITLE
Feature/ml-105:  add 'clear dataset cache' option

### PR DESCRIPTION
**Pull request by PierreLeveau**
**Overview**
Added a few features to make the training more robust and controllable (and my life easier):
* handle an argument to clear the dataset cache `--clear-dataset-cache`.
* added retries to handle disconnection when fetching the Yolo assets.
* added error catching for the yolo process.

**Testing**
Tested that the caching management was working for the NER and the object detection job.

### CODE REVIEW

**Comment by crsegerie** [comment]
As a general alternative, in order to memoize things in the past, I've used joblib, which is almost part of the standard library. You can check out an example, in the second comment: [example](https://stackoverflow.com/questions/16463582/memoize-to-disk-python-persistent-memoization) here

File: utils/huggingface/converters.py [hunk 1]
```diff
@@ -26,55 +27,58 @@ def kili_assets_to_hf_ner_dataset(
...
+                        "Authorization": f"X-API-Key: {api_key}",
+                    },
+                )
+                text = response.text
+                annotations = asset["labels"][0]["jsonResponse"][job_name]["annotations"]
+                sentences = sent_tokenize(text)
+                offset = 0
+                for sentence_tokens in TreebankWordTokenizer().span_tokenize_sents(
+                    sentences
+                ):
```
Comment (crsegerie) [inline utils/huggingface/converters.py:47]: I would put this loop in another function, to reduce the number of tabs.

File: utils/ultralytics/train.py [hunk 2]
```diff
@@ -43,15 +52,38 @@ def ultralytics_train_yolov5(
...
+            f'"{output_path}"',
+            *args_from_json,
+        ]
+        print(args)
+        subprocess.run(
+            args,
+            check=True,
+        )
+    except subprocess.CalledProcessError:
+        kili_print("YoloV5 training crashed")
```
Comment (crsegerie) [inline utils/ultralytics/train.py:83]: Not now, but later on we could implement a KiliError class and raise the error with the stack

**Comment by PierreLeveau** [comment]
>  As a general alternative, in order to memoize things in the past, I've used joblib, which is almost part of the standard library. You can check out an [example](https://stackoverflow.com/questions/16463582/memoize-to-disk-python-persistent-memoization), in the second comment: example here

For this, I agree that's a nice way to do caching for function call results. However here we need to save the assets in a readable file format (`.txt`, `.jpg`, ...) because hugging face and yolo require this file format to work.

File: utils/ultralytics/train.py (hunk 2 above)
Comment (PierreLeveau) [reply utils/ultralytics/train.py:83]: why not. In fact there is a mistake there: I forgot to raise the exception after the print. Doing it.

**Comment by PierreLeveau** [comment]
@crsegerie answered

**Comment by crsegerie** [comment]
lgtm!

**Pull request merged**